app/src/main/java/com/cs102/studybuddy/Register.java

```diff
@@ -0,0 +1,103 @@
+package com.cs102.studybuddy;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+
+public class Register extends AppCompatActivity {
+
+    private EditText regEmail, regPass, regConfirm, year, gender;
+    private Button signUp;
+    private FirebaseAuth mAuth;
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        // Check if user is signed in (non-null) and update UI accordingly.
+//        FirebaseUser currentUser = mAuth.getCurrentUser();
+//        if (currentUser != null){
+//            Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+//            startActivity(intent);
+//            finish();
+//        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```

Comment:
_**For future us**_: This also should be refactored

Reply:
I am not sure how can we refactor this. Please give more details. From what i know this method, while exists for different views, will have different functionality as you can see. But maybe I did not understand.

Reply:
By refactoring, I don't mean that in a generic way, but to extract code fragments to functions in order to keep the readability in place.

For example, the onComplete method of OnCompleteListener can be extracted to a function/method and with the support of [Lambda Expressions that were added in Java 8](https://www.w3schools.com/java/java_lambda.asp), we can get rid of the old ugly syntax.

This can also be applied to onClick method of OnClickListener, but again, this is just for readability so not a focus right now.